src/main/java/com/limechain/network/protocol/warp/DigestHelper.java

```diff
@@ -1,45 +1,47 @@
 package com.limechain.network.protocol.warp;
 
+import com.limechain.babe.consensus.BabeConsensusMessage;
 import com.limechain.babe.predigest.BabePreDigest;
+import com.limechain.babe.consensus.scale.BabeConsensusMessageReader;
 import com.limechain.babe.predigest.scale.PreDigestReader;
-import com.limechain.babe.state.EpochState;
 import com.limechain.network.protocol.warp.dto.ConsensusEngine;
 import com.limechain.network.protocol.warp.dto.DigestType;
 import com.limechain.network.protocol.warp.dto.HeaderDigest;
-import com.limechain.rpc.server.AppBean;
 import com.limechain.utils.scale.ScaleUtils;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
 
 import java.util.Arrays;
 import java.util.Optional;
 
 /**
  * Helper class for processing different types of header digests
  */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class DigestHelper {
-    private EpochState epochState;
 
-    public DigestHelper() {
-        this.epochState = AppBean.getBean(EpochState.class);
-    }
-
-    public void handleHeaderDigests(HeaderDigest[] headerDigests) {
-        if (headerDigests == null || headerDigests.length == 0) {
-            return;
-        }
-        updateEpochStateIfBabeConsensusMessageExists(headerDigests);
-        handleBabePreRuntimeDigest(headerDigests);
-    }
+//    public Map<DigestType, Digest> handleHeaderDigests(HeaderDigest[] headerDigests) {
+//        if (headerDigests == null || headerDigests.length == 0) {
+//            return null;
+//        }
+//
+//        Map<DigestType, Digest> digests = new HashedMap<>();
+//        getBabeConsensusMessage(headerDigests).ifPresent(d -> digests.put(DigestType.CONSENSUS_MESSAGE, d));
+//        getBabePreRuntimeDigest(headerDigests).ifPresent(d -> digests.put(DigestType.PRE_RUNTIME, d));
+//
+//        return digests;
+//    }
```

Comment:
Is this forgotten?

Reply:
No, I just committed a temporal state of the branch and didn't find how to put the PR on hold. It has been removed now, but there are new TODOs that will be fixed at https://github.com/LimeChain/Fruzhin/issues/593.